halogen/types.py

```diff
@@ -173,21 +175,25 @@ class Boolean(Type):
     def serialize(self, value, **kwargs):
         return bool(value) if value is not None else None
 
-    def deserialize(self, value, **kwargs):
-        try:
-            value = int(value)
+    def deserialize(self, value: Union[str, int, bool, None], **kwargs):
+        if isinstance(value, int):
             value = bool(value) if value == 1 or value == 0 else None
-        except ValueError:
-            str_value = str(value).lower()
-            if str_value == "true":
-                value = True
-            elif str_value == "false":
-                value = False
 
-        if not isinstance(value, bool):
-            raise ValueError("'{val}' is not an 1 or 0 and true or false".format(val=value))
+        if isinstance(value, str):
+            try:
+                value = int(value)
```

Comment:
This will create unexpected errors, you should first check whether value is numerical.